shared/actions/gregor.js

```diff
@@ -63,31 +63,33 @@ const setupEngineListeners = () => {
 
   // Filter this firehose down to the two systems we care about: "git", and "kbfs.favorites"
   // If ever you want to get OOBMs for a different system, then you need to enter it here.
-  RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
-    .then(response => {
-      logger.info('Registered gregor listener')
-    })
-    .catch(error => {
-      logger.warn('error in registering gregor listener: ', error)
-    })
+  engine().actionOnConnect('registerGregorFirehose', () =>
+    RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
```

Comment:
I think the flow error here is that you're doing `() => RPC...`, which creates a function that _returns_ the expression (which isn't an action), whereas you want `() => { RPC... }`, which creates a function that evaluates the expression but returns undefined.